docs/release-notes.d/00readme.md

```diff
@@ -0,0 +1,28 @@
+# Pending release notes directory
+
+This directory contains release note entries that have not been merged
+into the main release-notes.md document.
+
+Generally a release note entry should be created for changes that:
```

Comment:
Should we have some github label that will tell that change will require or is providing release note?

Reply:
Maybe a CI job that marks changes based on some rules, and then we can override that if it gets it wrong.